### DISCUSSION TITLE
Changing Default Supply Voltage Value?

### DISCUSSION THREAD

**Discussion by dalyles** [The-OpenROAD-Project/OpenROAD-flow-scripts#1702] [Q&A]
Where can we modify the supply voltage?
It looks like it's at 1.8V by default.

I want to do runs on the same circuit and sweep with various values for VDD.

Thanks!

**Comment by maliberty** [comment]
Would set_pvt in SDC suffice?

**Reply by dalyles** [reply]
I see that in the sdc.tcl file.
Where (which file) would we set the values?

And also where would we be able to find the default values that are set in set_pvt?
I do not see this in the sdc.tcl file.

Thanks

**Reply by dalyles** [reply]
Following up on this

**Reply by maliberty** [reply]
The voltage depends on the .lib.   For example is sky130hd:
```
 operating_conditions ("tt_025C_1v80") {
        voltage : 1.8000000000;
        process : 1.0000000000;
        temperature : 25.000000000;
        tree_type : "balanced_tree";
    }
```
while in asap7
```
  operating_conditions (PVT_0P7V_25C) {
    process : 1;
    temperature : 25;
    voltage : 0.7;
  }
```
You would add the command to whatever sdc file you are using based on SDC_FILE

**Comment by dalyles** [comment]
Ok so let me understand this.

All of the runs PVT come directly from Liberty files and we do not have
such an option to modify our PVT choices **without** modifying any sensitive
codes like TCL etc.? In other words, there is no user friendly way to tweak
such parameters?

Thanks

**Reply by maliberty** [reply]
I don't understand you comment - I suggested an SDC command.

**Comment by dalyles** [comment]
Is it possible to set those values in constraint.sdc or config.mk?

**Reply by rovinski** [reply]
The best way to do this would be to set up a corner for each voltage you want to sweep. Look up the `CORNERS` variable. [ASAP7 has an example](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/blob/65ab85e8c5d416f14fbd3b429099b14b4ed35cff/flow/platforms/asap7/config.mk), although it doesn't seem well documented.

The key thing to note is you have to have a lib file characterized at each of the characterization points you're looking to sweep.
